### PR TITLE
Add purview workflow

### PR DESCRIPTION
**Pull request by CordeliaGrey**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/Directory.Build.Common.targets [hunk 1]
```diff
@@ -32,7 +32,7 @@
       <ValidateReleaseNotesScriptPath Condition=" '$(ValidateReleaseNotesScriptPath)'=='' ">$(MSBuildThisFileDirectory)common/scripts/Verify-ChangeLog.ps1</ValidateReleaseNotesScriptPath>
       <ChangeLogPath>$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory)/../))CHANGELOG.md</ChangeLogPath>
     </PropertyGroup>
-    <Exec ContinueOnError="true" ConsoleToMSBuild="true" StandardOutputImportance="High" Command="$(PowerShellExe) -NoProfile -NonInteractive -executionpolicy Unrestricted -File $(ValidateReleaseNotesScriptPath) -ChangeLogLocation $(ChangeLogPath) -VersionString $(_VersionInProject)">
+    <Exec ContinueOnError="true" ConsoleToMSBuild="true" StandardOutputImportance="High" Command="$(PowerShellExe) -NoProfile -NonInteractive -executionpolicy Unrestricted -Command $(ValidateReleaseNotesScriptPath) -ChangeLogLocation $(ChangeLogPath) -VersionString $(_VersionInProject)">
```
Comment (jsquire) [inline eng/Directory.Build.Common.targets:35]: Can you help me understand why this change is being proposed?

**Review by jsquire — CHANGES_REQUESTED** [review]

File: eng/Directory.Build.Common.targets (hunk 1 above)
Comment (CordeliaGrey) [reply eng/Directory.Build.Common.targets]: This command doesn't work in my local so I made the changes, that's an unexpected commit, I have removed this changes.

**Review by jsquire — CHANGES_REQUESTED** [review]

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+- CRUD of workflow.
+- Submit user request and monitor workflow run.
+- View and respond to workflow tasks.
+
+
+**Please rely heavily on the [service's documentation][product_documentation]**
+
+## Getting started
+
+This section should include everything a developer needs to do to install and create their first client connection *very quickly*.
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/README.md:15]: This section was reserved for the sdk download address, I have creates the issue for tracking completing README, here is issue link: https://github.com/Azure/azure-sdk-for-net/issues/33756

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    //<client-id>, <tenant-id> are the client ID and tenant ID of the AAD application.
+    //<user-name>,<password> are the value of user name and password of an AAD user.
+    TokenCredential usernamePasswordCredential = new UsernamePasswordCredential(<client-id>, <tenant-id>, <user-name>,<password>, null);
+    var client = new PurviewWorkflowServiceClient(new Uri("https://<purview-account-name>.purview.azure.com"), usernamePasswordCredential)
+```
+
+
+## Key concepts
+
+The *Key concepts* section should describe the functionality of the main classes. Point out the most important and useful classes in the package (with links to their reference pages) and explain how those classes work together. Feel free to use bulleted lists, tables, code blocks, or even diagrams for clarity.
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/README.md:37]: Already tracked in issue, issue link: https://github.com/Azure/azure-sdk-for-net/issues/33756

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+var credential = new DefaultAzureCredential();
+var client = new PurviewWorkflowServiceClient(endpoint, credential);
+
+Guid workflowId = new Guid("ba25ed0e-3364-4e8e-8385-c60e12f3e342");
+Response result = client.GetWorkflow(workflowId);
+```
+
+## Troubleshooting
+
+Describe common errors and exceptions, how to "unpack" them if necessary, and include guidance for graceful handling and recovery.
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/README.md:75]: Already tracked in issue, issue link: https://github.com/Azure/azure-sdk-for-net/issues/33756

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/Generated/PurviewWorkflowServiceClient.cs [hunk 5]
```diff
@@ -0,0 +1,1277 @@
...
+        /// <returns> The <see cref="AsyncPageable{T}"/> from the service containing a list of <see cref="BinaryData"/> objects. Details of the body schema for each item in the collection are in the Remarks section below. </returns>
+        /// <include file="Docs/PurviewWorkflowServiceClient.xml" path="doc/members/member[@name='GetWorkflowsAsync(RequestContext)']/*" />
+        public virtual AsyncPageable<BinaryData> GetWorkflowsAsync(RequestContext context = null)
+        {
+            return GetWorkflowsImplementationAsync("PurviewWorkflowServiceClient.GetWorkflows", context);
+        }
+
+        private AsyncPageable<BinaryData> GetWorkflowsImplementationAsync(string diagnosticsScopeName, RequestContext context)
+        {
+            return PageableHelpers.CreateAsyncPageable(CreateEnumerableAsync, ClientDiagnostics, diagnosticsScopeName);
```
Comment (pshao25) [inline sdk/purview/Azure.Analytics.Purview.Workflows/src/Generated/PurviewWorkflowServiceClient.cs:720]: To resolve build error, you could sync the latest code.

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Release History
```
Comment (pshao25) [inline sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md:1]: Please add your changelog content.

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+### Prerequisites
+
+You must have an [Azure subscription][azure_subscription] and a [Purview resource][purview_resource] to use this package.
+
+### Authenticate the client
+
+Since the Workflow service uses an Azure Active Directory (AAD) bearer token for authentication and identification, an email address should be encoded into the token to allow for notification when using Workflow. It is recommended that the [Azure Identity][azure_identity] library be used  with a the [UsernamePasswordCredential][username_password_credential]. Before using the [Azure Identity][azure_identity] library with Workflow, [an application][app_registration] should be registered and used for the clientId passed to the [UsernamePasswordCredential][username_password_credentail].
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_CreatePurviewWorkflowClient
```
Comment (pshao25) [inline sdk/purview/Azure.Analytics.Purview.Workflows/README.md:20]: Run `eng\scripts\Update-Snippets.ps1` to fill the real content into this snippet.

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs [hunk 8]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "CancellationToken can be passed through RequestOptions")]
```
Comment (pshao25) [inline sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs:6]: Why we need this suppression?

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+// Replace with test project/test project public key and uncomment to make internal members visible to
```
Comment (pshao25) [inline sdk/purview/Azure.Analytics.Purview.Workflows/src/properties/AssemblyInfo.cs:6]: We could remove the comments.

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#consuming-long-running-operations-using-operationt) |
+[Handling failures](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#reporting-errors-requestfailedexception) |
+[Diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md) |
+[Mocking](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#mocking) |
+[Client lifetime](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/)
+<!-- CLIENT COMMON BAR -->
+
+## Examples
+
+You can familiarize yourself with different APIs using [Samples](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/purview/Azure.Analytics.Purview.Workflows/samples).
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/README.md:25]: Thanks for your feedback, I've already added examples in README.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
+# Submit user requests
+This sample shows how to submit user request.
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md]: A user request describes user ask to do operation(s) on Purview, user need to submit user request to trigger a runing instance of a workflow.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
+# Workflow runs
+
+This sample shows how to cancel a workflow run.
+
+To get started, make sure you have satisfied all the prerequisites and got all the resources required by [README][README].
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md:5]: Please help me understand why we can't include the information needed to make this a self-contained sample.   Asking developers to click around to find information adds a level of difficulty when learning to use the SDK.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    TokenCredential usernamePasswordCredential = new UsernamePasswordCredential(<client-id>, <tenant-id>, <user-name>,<password>, null);
+    var client = new PurviewWorkflowServiceClient(new Uri("https://<purview-account-name>.purview.azure.com"), usernamePasswordCredential)
+```
+
+
+## Approve workflow task
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_ApproveWorkflowTask
+
+    Guid taskId = new Guid("b129fe16-72d3-4994-9135-b997b9be46e0");
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md:23]: I see this resolved, but I do not see anything that actually addresses the comment.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+This sample shows how to create and get a workflow.
+
+To get started, make sure you have satisfied all the prerequisites and got all the resources required by [README][README].
+
+## Create a WorkflowClient
+
+To create a new `WorkflowClient` you need the endpoint, apiVersion, and credentials from your resource. In the sample below you'll use UsernamePasswordCredential to authenticate.
+You can set `endpoint`, `username`, `password` etc. based on an environment variable, a configuration setting, or any way that works for your application.
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_CreatePurviewWorkflowClient
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md:11]: Please help me understand where the corresponding snippet definition is.   It does not appear to be in the tests.

**Review by jsquire — CHANGES_REQUESTED** [review]

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md]: I missed some files in the last commit, sorry for misleading you,  I've added the comment in the sample to explain how to get the task id.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md]: I missed some files in the last commit, sorry for misleading you, I've added the link into this file of a specific section of README, the prerequisites is not a long list but I need to explain why we need a certain resource, I put this explanation in README.

File: sdk/purview/Azure.Analytics.Purview.Workflows/tests/WorkflowsClientTestEnvironment.cs [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+namespace Azure.Analytics.Purview.Workflows.Tests
+{
+    public class WorkflowsClientTestEnvironment : TestEnvironment
+    {
+        private TokenCredential _usernamePasswordcredential;
+
+        public WorkflowsClientTestEnvironment()
+        {
+        }
+        public Uri Endpoint => new(GetRecordedVariable("Workflow_ENDPOINT"));
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/tests/WorkflowsClientTestEnvironment.cs]: Fixed, thanks.

File: sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+using Azure.Core;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Analytics.Purview.Workflows.Tests.Samples
+{
+    public partial class WorkflowsSamples: SamplesBase<WorkflowsClientTestEnvironment>
+    {
+        /* please refer to https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/template/Azure.Template/tests/Samples/Sample1.HelloWorld.cs to write samples. */
+        #region Snippet:Azure_Analytics_Purview_Workflows_Scenario
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs]: Fixed, thanks.

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
+# Azure Purview Workflow client library for .NET
+
+Workflows are automated, repeatable business processes that users can create within Microsoft Purview to validate and orchestrate CUD (create, update, delete) operations on their data entities. Enabling these processes allow organizations to track changes, enforce policy compliance, and ensure quality data across their data landscape.
+
+Use the client library for Purview Workflow to:
+- Managing workflows
+- Submitting user requests and monitoring workflow runs
+- Viewing and responding to workflow tasks
+
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/README.md:9]:
```suggestion
- Manage workflows
- Submit user requests and monitoring workflow runs
- View and respond to workflow tasks

```

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+```C# Snippet:Azure_Analytics_Purview_Workflows_SubmitUserRequests
+string request = "{\"operations\":[{\"type\":\"CreateTerm\",\"payload\":{\"glossaryTerm\":{\"name\":\"term\",\"anchor\":{\"glossaryGuid\":\"20031e20-b4df-4a66-a61d-1b0716f3fa48\"},\"status\":\"Approved\",\"nickName\":\"term\"}}}],\"comment\":\"Thanks!\"}";
+
+Response submitResult = await client.SubmitUserRequestsAsync(RequestContent.Create(request));
+```
+
+### Approve workflow task
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_ApproveWorkflowTask
+// This taskId is an existing workflow task's id, user could get workflow tasks by calling GetWorkflowTasksAsync API.
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/README.md:60]:
```suggestion
// This taskId represents an existing workflow.  The id can be obtained by calling GetWorkflowTasksAsync.
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+string request = "{\"comment\":\"Thanks!\"}";
+
+Response approveResult = await client.ApproveApprovalTaskAsync(taskId, RequestContent.Create(request));
+```
+
+<!-- LINKS -->
+[product_documentation]: https://learn.microsoft.com/azure/purview/concept-workflow
+[azure_subscription]: https://azure.microsoft.com/free/dotnet/
+[purview_resource]: https://docs.microsoft.com/azure/purview/create-catalog-portal
+[azure_identity]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/README.md:72]:
```suggestion
[azure_identity]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity/README.md
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md [hunk 20]
```diff
@@ -0,0 +1,43 @@
+# Create and get a Workflow
+This sample shows how to create and get a workflow.
+
+To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and got all the resources required by [Authenticate the client][authenticate_the_client].
+
+
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md:6]:
```suggestion

```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md [hunk 21]
```diff
@@ -0,0 +1,43 @@
+# Create and get a Workflow
+This sample shows how to create and get a workflow.
+
+To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and got all the resources required by [Authenticate the client][authenticate_the_client].
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md:4]:
```suggestion
To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and have obtained all of the resources required by [Authenticate the client][authenticate_the_client].
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md [hunk 22]
```diff
@@ -0,0 +1,30 @@
+# Submit user requests
+This sample shows how to submit a user request.
+
+To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and got all the resources required by [Authenticate the client][authenticate_the_client].
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md:4]:
```suggestion
To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and have obtained all of the resources required by [Authenticate the client][authenticate_the_client].
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md [hunk 23]
```diff
@@ -0,0 +1,34 @@
+# Workflow tasks
+
+This sample shows how to approve s workflow task.
+
+To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and got all the resources required by [Authenticate the client][authenticate_the_client].
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md:5]:
```suggestion
To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and have obtained all of the resources required by [Authenticate the client][authenticate_the_client].
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_ApproveWorkflowTask
+// This taskId is an existing workflow task's id, user could get workflow tasks by calling GetWorkflowTasksAsync API.
+Guid taskId = new Guid("b129fe16-72d3-4994-9135-b997b9be46e0");
+
+string request = "{\"comment\":\"Thanks!\"}";
+
+Response approveResult = await client.ApproveApprovalTaskAsync(taskId, RequestContent.Create(request));
+```
+[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md:33]:
```suggestion

[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md [hunk 25]
```diff
@@ -0,0 +1,34 @@
+# Workflow runs
+
+This sample shows how to cancel a workflow run.
+
+To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and got all the resources required by [Authenticate the client][authenticate_the_client].
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md:5]:
```suggestion
To get started, make sure you have satisfied all the [Prerequisites][prerequisites] and have obtained all of the resources required by [Authenticate the client][authenticate_the_client].
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_CancelWorkflowRun
+// This workflowRunId is an existing workflow run's id, user could get workflow runs by calling GetWorkflowRunsAsync API.
+Guid workflowRunId = new Guid("4f8d70c3-c09b-4e56-bfd1-8b86c79bd4d9");
+
+string request = "{\"comment\":\"Thanks!\"}";
+
+Response cancelResult = await client.CancelWorkflowRunAsync(workflowRunId, RequestContent.Create(request));
+```
+[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample4_WorkflowRuns.md:33]:
```suggestion

[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+## Submit a user request
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_SubmitUserRequests
+string request = "{\"operations\":[{\"type\":\"CreateTerm\",\"payload\":{\"glossaryTerm\":{\"name\":\"term\",\"anchor\":{\"glossaryGuid\":\"20031e20-b4df-4a66-a61d-1b0716f3fa48\"},\"status\":\"Approved\",\"nickName\":\"term\"}}}],\"comment\":\"Thanks!\"}";
+
+Response submitResult = await client.SubmitUserRequestsAsync(RequestContent.Create(request));
+```
+[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample2_SubmitUserRequests.md:29]:
```suggestion

[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+
+## Get a Workflow
+
+```C# Snippet:Azure_Analytics_Purview_Workflows_GetWorkflow
+// This workflowId is an existing workflow's id, user could create a new workflow by calling CreateOrReplaceWorkflowAsync API or list workflows by calling GetWorkflowsAsync API.
+Guid workflowId = new Guid("8af1ecae-16ee-4b2d-8972-00d611dd2f99");
+
+Response getResult = await client.GetWorkflowAsync(workflowId);
+```
+[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md:41]:
```suggestion

[prerequisites]: https://github.com/Azure/azure-sdk-for-net/sdk/purview/Azure.Analytics.Purview.Workflows/README.md#Prerequisites
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs [hunk 29]
```diff
@@ -0,0 +1,107 @@
...
+            JsonElement getBodyJson = jsonDocument.RootElement;
+            Assert.AreEqual(workflowId.ToString(), getBodyJson.GetProperty("id").GetString());
+        }
+
+        private static BinaryData GetContentFromResponse(Response r)
+        {
+            // Workaround azure/azure-sdk-for-net#21048, which prevents .Content from working when dealing with responses
+            // from the playback system.
+
+            MemoryStream ms = new MemoryStream();
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs:102]:
```suggestion
            using MemoryStream ms = new MemoryStream();
```

File: sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs [hunk 30]
```diff
@@ -0,0 +1,21 @@
...
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Analytics.Purview.Workflows.Tests.Samples
+{
+    public partial class WorkflowsSamples: SamplesBase<WorkflowsClientTestEnvironment>
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/tests/Samples/Sample1_HelloWorld.cs:16]: Fixed, thanks.

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md (hunk 24 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample3_WorkflowTasks.md:33]: May I ask what's the difference between your comment and original?

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/Generated/PurviewWorkflowServiceClient.cs (hunk 5 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/src/Generated/PurviewWorkflowServiceClient.cs]: Fixed, thanks for your feedback!

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md (hunk 6 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md:1]: Done

File: sdk/purview/Azure.Analytics.Purview.Workflows/README.md (hunk 7 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/README.md]: Done

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs (hunk 8 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs:6]: It is generated automatically

File: sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md [hunk 31]
```diff
@@ -0,0 +1,39 @@
...
...val\",\"inputs\":{\"parameters\":{\"approvalType\":\"PendingOnAll\",\"title\":\"ApprovalRequestforCreateGlossaryTerm\",\"assignedTo\":[\"eece94d9-0619-4669-bb8a-d6ecec5220bc\"]}},\"runAfter\":{}},\"Condition\":{\"type\":\"If\",\"expression\":{\"and\":[{\"equals\":[\"@outputs('Startandwaitforanapproval')['body/outcome']\",\"Approved\"]}]},\"actions\":{\"Createglossaryterm\":{\"type\":\"CreateTerm\",\"runAfter\":{}},\"Sendemailnotification\":{\"type\":\"EmailNotification\",\"inputs\":{\"parameters\":{\"emailSubject\":\"GlossaryTermCreate-APPROVED\",\"emailMessage\":\"YourrequestforGlossaryTerm@{triggerBody()['request']['term']['name']}isapproved.\",\"emailRecipients\":[\"@{triggerBody()['request']['requestor']}\"]}},\"runAfter\":{\"Createglossaryterm\":[\"Succeeded\"]}}},\"else\":{\"actions\":{\"Sendrejectemailnotification\":{\"type\":\"EmailNotification\",\"inputs\":{\"parameters\":{\"emailSubject\":\"GlossaryTermCreate-REJECTED\",\"emailMessage\":\"YourrequestforGlossaryTerm@{triggerBody()['request']['term']['name']}isrejected.\",\"emailRecipients\":[\"@{triggerBody()['request']['requestor']}\"]}},\"runAfter\":{}}}},\"runAfter\":{\"Startandwaitforanapproval\":[\"Succeeded\"]}}}}}";
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/samples/Sample1_Workflows.md]: This is a highly customized payload, user could copy the payload from UI and paste it in code, here using string just provide a sample.

**Review by jsquire — COMMENTED** [review]
I left a couple of minor formatting nits.   The only thing blocking is the change log, which would fail pipeline validation.

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md [hunk 32]
```diff
@@ -0,0 +1,17 @@
...
+
+### Features Added
+
+### Breaking Changes
+
+### Bugs Fixed
+
+### Other Changes
+
+## 1.0.0-beta.1
```
[the remark this answers is not in the document]
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md:13]: Fixed, thanks

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md [hunk 33]
```diff
@@ -0,0 +1,17 @@
+# Release History
+
+## 1.0.0-beta.2 (Unreleased)
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md:3]: This needs to match the version in the `.csproj` file, which still reads `1.0.0-beta.2`

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md [hunk 34]
```diff
@@ -0,0 +1,17 @@
...
+
+### Features Added
+
+### Breaking Changes
+
+### Bugs Fixed
+
+### Other Changes
+
+## 1.0.0-beta.1
```
Comment (jsquire) [inline sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md:13]: Can you point me to where this version was released in NuGet?   It looks like we have a single changelog entry for `1.0.0-beta.1` and remove `beta.2`

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md (hunk 33 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md]: Fixed, thanks

File: sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md (hunk 34 above)
Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/CHANGELOG.md]: Fixed, thanks

**Review by jsquire — APPROVED** [review]
Project assets look good to me.   The build is failing because Analyze is flagging `azure-purview-workflow` an invalid product for the samples metadata.    I'm not familiar with where the legal values come from, but @danieljurek likely is.

In the meantime,  you may want to try `azure-purview` and see if the more generalized category is what they're looking for.

**Comment by danieljurek** [comment]
The taxonomy site is here: https://taxonomy.learn.microsoft.com/TaxonomyServiceAdminPage/#/taxonomy/detail/2023-01-27T09:00:06.1263332Z!5a3ccca6-19ce-4f32-b516-88038fa259ae/product

I think the correct value is `microsoft-purview` but this should be verified with PM on the service team.

File: sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs (hunk 8 above)
Comment (pshao25) [reply sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs:6]: It is not generated. All the generated stuff will be under Gnerated folder.

Comment (CordeliaGrey) [reply sdk/purview/Azure.Analytics.Purview.Workflows/src/GlobalSuppressions.cs]: Fixed, thanks
